RepoDb.Core/RepoDb/ClassMap.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using System.Linq.Expressions;
+using RepoDb.Interfaces;
+
+namespace RepoDb
+{ 
+    public class ClassMap<T> where T : class
+    {
+        public Type EntityType { get; }
+
+        public ClassMap()
+        {
+            EntityType = typeof(T);
+        }
+        
+        protected ClassMap<T> Table(string table)
+        {
+            ClassMapper.Add<T>(table);
+            return this;
+        }
+        
+        protected IIdentityOptions<T> Id(Expression<Func<T, object>> expression)
+        {
+            IdentityMapper.Add<T>(expression);
+            return new IdentityOptions<T>(expression);
+        }
+
+        protected IPrimaryOptions<T> Primary(Expression<Func<T, object>> expression)
+        {
+            PrimaryMapper.Add<T>(expression);
+            return new PrimaryOptions<T>(expression);
+        }
+
+        protected IPropertyOptions<T> Map(Expression<Func<T, object>> expression)
```

Comment:
And for this one, the name should be `Property`? Or, please assess.

If find it a little confusing to do it like this.

```csharp
CustomerMap
	.Table("[dbo].[Customer]")
	.Map(e => e.FirstName)
		.Column("FName");
CustomerMap
	.Map(e => e.LastName)
		.Column("LName");
CustomerMap
	.Map(e => e.DateOfBirth)
		.DbType(DbType.DateTime2);
```

But this seems to be okay.

```csharp
CustomerMap
	.Table("[dbo].[Customer]")
	.Property(e => e.FirstName)
		.Column("FName");
CustomerMap
	.Property(e => e.LastName)
		.Column("LName");
CustomerMap
	.Property(e => e.DateOfBirth)
		.DbType(DbType.DateTime2);
```

Reply:
I will handle this failing Integration Tests, I do not why it fails with this PR. I will try to see running this locally. The test result can be found [here](https://ci.appveyor.com/project/mikependon/repodb-qksas/build/tests).

Reply:
ClassMaps should be used more like this:

public class CatClassMap : ClassMap<Cat>
{
    public CatClassMap()
    {
        Primary(x=> x.Id);
        Property(x=> x.Name).Column("Silver");
    }
}